ui/src/map/layers/geometry/geometry-alignment-layer.ts

```diff
@@ -108,78 +109,80 @@ type AlignmentWithLinking = {
 };
 
-async function getPlanLayoutAlignmentsWithLinking(
-    planLayout: GeometryPlanLayout,
-    publishType: PublishType,
-    changeTime: TimeStamp,
+function getAlignmentsWithLinking(
+    alignments: PlanLayoutAlignment[],
+    linkedAlignmentIds: GeometryAlignmentId[],
     resolution: number,
-): Promise<AlignmentWithLinking[]> {
-    const planLayoutWithGeometry =
-        planLayout.planDataType == 'STORED'
-            ? await getTrackLayoutPlan(planLayout.planId, changeTime)
-            : planLayout;
-
-    if (!planLayoutWithGeometry) {
-        return [];
-    }
-
-    const linkedAlignmentIds =
-        planLayout.planDataType === 'STORED'
-            ? await getLinkedAlignmentIdsInPlan(planLayout.planId, publishType)
-            : [];
-
-    return (
-        planLayoutWithGeometry.alignments
-            // Include alignments from original layout only
-            .filter((a1) => planLayout.alignments.some((a2) => a1.header.id === a2.header.id))
-            .map((alignment) => {
-                const points = alignment.polyLine?.points || [];
-                return {
-                    header: alignment.header,
-                    points: filterLayoutPoints(toMapAlignmentResolution(resolution), points),
-                    segmentMValues: alignment.segmentMValues,
-                    linked: alignment.header.id
-                        ? linkedAlignmentIds.includes(alignment.header.id)
-                        : false,
-                };
-            })
-    );
+): AlignmentWithLinking[] {
+    return alignments.map((alignment) => {
+        const points = alignment.polyLine?.points || [];
+        return {
+            header: alignment.header,
+            points: filterLayoutPoints(toMapAlignmentResolution(resolution), points),
+            segmentMValues: alignment.segmentMValues,
+            linked: alignment.header.id ? linkedAlignmentIds.includes(alignment.header.id) : false,
+        };
+    });
 }
 
 let newestLayerId = 0;
 
+type PlanAlignments = {
+    planId: GeometryPlanId;
+    alignments: AlignmentWithLinking[];
+};
 export function createGeometryAlignmentLayer(
     existingOlLayer: VectorLayer<VectorSource<LineString>> | undefined,
     selection: Selection,
     publishType: PublishType,
     changeTimes: ChangeTimes,
     resolution: number,
+    manuallySetPlan?: GeometryPlanLayout,
 ): MapLayer {
     const layerId = ++newestLayerId;
 
+    const visibleAlignmentIds = manuallySetPlan
+        ? manuallySetPlan.alignments.map((a) => a.header.id)
+        : selection.visiblePlans.flatMap((p) => p.alignments);
     const vectorSource = existingOlLayer?.getSource() || new VectorSource();
     const olLayer = existingOlLayer || new VectorLayer({ source: vectorSource });
 
-    const changeTime = getMaxTimestamp(
-        changeTimes.layoutReferenceLine,
-        changeTimes.layoutLocationTrack,
+    let inFlight = true;
+
+    const plansPromise: Promise<GeometryPlanLayout[]> = manuallySetPlan
```

Comment:
Nämä layerin datankeräilyt on edelleen vähän kökköiset. Kokeilen huomenna vähän siistiä näitä.